cmd/cosign/cli/sign/sign.go

```diff
@@ -66,18 +66,13 @@ const TagReferenceMessage string = `WARNING: Image reference %s uses a tag, not
     images by tag will be removed in a future release.
 `
 
-func ShouldUploadToTlog(ctx context.Context, ko options.KeyOpts, ref name.Reference, force, noTlogUpload bool) bool {
+func ShouldUploadToTlog(ctx context.Context, ko options.KeyOpts, ref name.Reference, force, tlogUpload bool) bool {
 	// We are forcing publishing to the Tlog.
 	if force {
 		return true
 	}
-	// Check whether to not upload Tlog.
-	if noTlogUpload {
-		return false
-	}
-
-	// If we aren't using keyless signing, we can skip the tlog
-	if !keylessSigning(ko) {
+	// If we aren't using keyless signing and --tlog-upload=false, return
+	if !keylessSigning(ko) && !tlogUpload {
```

Comment:
keyless = true, tlog = true - upload
keyless = true, tlog = false - don't upload
keyless = false, tlog = true - upload
keyless = false, tlog = false - don't upload

So keyless doesn't matter then, correct? We can just set tlog-upload's default to true.

Reply:
Ah, I guess a problem here is we don’t want to default to be true, for the managed key cases. So we should check keyless. 

In the code, I think we still need to handle the case of keyless = true, and tlog = false

Reply:
> Ah, I guess a problem here is we don’t want to default to be true, for the managed key cases. 

Yah, exactly! 

> I think we still need to handle the case of keyless = true, and tlog = false

in that case, verification would always fail, so i feel like we shouldn't allow users to sign that way to begin with.

Reply:
That'll change if they're allowed to get a user-supplied timestamp, but they'd be using cosign libraries at that point and not CLIs.
I think this checks fine, but you might want to add a comment saying that we must upload for keyless.

Reply:
sg, added the comment!